sources/api/schnauzer/src/helpers.rs

```diff
@@ -422,6 +400,163 @@ pub fn default(
+/// number for a given region.
+///
+/// This helper takes a single initial parameter, "account" or "region". The
+/// second parameter is the region.
+///
+/// # Fallback
+///
+/// A map of region to ECR account ID is maintained herein. But if we do not
+/// have the region in our map, a fallback region and account number are
+/// returned. This would allow a version of Bottlerocket to run in a new region
+/// before this map has been updated, for example.
+///
+/// # Example
+///
+/// In this example the account number for the region will be returned.
+/// `{{ ecr "account" settings.aws.region }}`
+///
+/// In this example the region will either be returned, or if it happens to be
+/// a region for which we don't have an ECR repo, the fallback region will be
+/// returned.
+/// `{{ ecr "region" settings.aws.region }}`
+pub fn ecr(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting ecr helper");
+    let template_name = template_name(renderctx);
+    if helper.params().len() != 2 {
+        return Err(RenderError::from(
+            error::TemplateHelperError::IncorrectNumberOfParams {
+                expected: 4,
+                received: helper.params().len(),
+                helper: helper.name().to_string(),
+                template: template_name.to_owned(),
+            },
+        ));
+    }
+
+    // get the first parameter, which should be either 'account' or 'region'.
+    // convert this to a boolean is_account. error if param is bad.
+    let account_or_region_param = get_param(helper, 0)?;
+    let account_or_region =
+        account_or_region_param
+            .as_str()
+            .with_context(|| error::InvalidTemplateValue {
+                expected: "string",
+                value: account_or_region_param.to_owned(),
+                template: template_name.to_owned(),
+            })?;
+    let is_account = match account_or_region {
+        "account" => true,
+        "region" => false,
+        _ => {
+            return Err(RenderError::from(error::TemplateHelperError::EcrParam {
+                value: account_or_region.to_owned(),
+            }))
+        }
+    };
+
+    // get the second parameter, which is probably given by the template value
+    // settings.aws.region. regardless, we expect it to be a string.
+    let aws_region = get_param(helper, 1)?;
+    let aws_region = aws_region.as_str().ok_or_else(|| {
+        RenderError::from(error::TemplateHelperError::EcrRegion {
+            value: aws_region.to_owned(),
+            template: template_name.to_owned(),
+        })
+    })?;
+
+    // get a mapping of regions to their ECR accounts. find the pair for our
+    // region, or the fallback account and region if our region is not found.
+    let ecr_map = ecr_map();
+    let ecr_region_and_account = match ecr_map.get(aws_region) {
```

Comment:
```suggestion
    let (region, account) = match ecr_map.get(aws_region) {
```
Then we shouldn't need the `.0` and `.1` below.